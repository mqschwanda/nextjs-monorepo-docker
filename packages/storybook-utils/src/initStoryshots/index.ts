import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
