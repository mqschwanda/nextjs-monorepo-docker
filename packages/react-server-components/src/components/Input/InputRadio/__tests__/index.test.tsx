import { render } from '@testing-library/react';
import { InputRadio } from '@';
import initStoryshots from '@/__tests__/initStoryshots';

initStoryshots({
  failOnReactKeyError: false,
  storyKindRegex: /^@mqs\/react-server-components\/components\/Input\/InputRadio$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<InputRadio />', () => {
      it('renders', () => {
        const testId = InputRadio.name;

        const { getByTestId, asFragment } = render((
          <InputRadio
            className='test'
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
