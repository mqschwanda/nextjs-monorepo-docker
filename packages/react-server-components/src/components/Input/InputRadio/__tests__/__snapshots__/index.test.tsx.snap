// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <InputRadio /> renders 1`] = `
<DocumentFragment>
  <input
    class="radio radio-primary radio-md test"
    data-testid="InputRadio"
    type="radio"
  />
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputRadio Input Radio Color Example 1`] = `
[
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      accent
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-accent radio-md"
          data-testid="InputRadio"
          name="InputRadio-accent"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-accent radio-md"
          data-testid="InputRadio"
          name="InputRadio-accent"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      error
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-error radio-md"
          data-testid="InputRadio"
          name="InputRadio-error"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-error radio-md"
          data-testid="InputRadio"
          name="InputRadio-error"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      info
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-info radio-md"
          data-testid="InputRadio"
          name="InputRadio-info"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-info radio-md"
          data-testid="InputRadio"
          name="InputRadio-info"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      primary
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-primary radio-md"
          data-testid="InputRadio"
          name="InputRadio-primary"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-primary radio-md"
          data-testid="InputRadio"
          name="InputRadio-primary"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      secondary
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-secondary radio-md"
          data-testid="InputRadio"
          name="InputRadio-secondary"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-secondary radio-md"
          data-testid="InputRadio"
          name="InputRadio-secondary"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      success
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-success radio-md"
          data-testid="InputRadio"
          name="InputRadio-success"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-success radio-md"
          data-testid="InputRadio"
          name="InputRadio-success"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      warning
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-warning radio-md"
          data-testid="InputRadio"
          name="InputRadio-warning"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-warning radio-md"
          data-testid="InputRadio"
          name="InputRadio-warning"
          type="radio"
        />
      </label>
    </div>
  </div>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputRadio Input Radio Example 1`] = `
<div
  className="form-control w-[200px]"
  data-testid="FormControl"
>
  <label
    className="label"
    data-testid="Label"
  >
    Example
  </label>
  <div
    className="form-control"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        Option 1
      </span>
      <input
        className="radio radio-primary radio-md"
        data-testid="InputRadio"
        name="InputRadioExample"
        type="radio"
      />
    </label>
  </div>
  <div
    className="form-control"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        Option 2
      </span>
      <input
        className="radio radio-primary radio-md"
        data-testid="InputRadio"
        name="InputRadioExample"
        type="radio"
      />
    </label>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputRadio Input Radio Size Example 1`] = `
[
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      lg
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-primary radio-lg"
          data-testid="InputRadio"
          name="InputRadio-lg"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-primary radio-lg"
          data-testid="InputRadio"
          name="InputRadio-lg"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      md
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-primary radio-md"
          data-testid="InputRadio"
          name="InputRadio-md"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-primary radio-md"
          data-testid="InputRadio"
          name="InputRadio-md"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      sm
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-primary radio-sm"
          data-testid="InputRadio"
          name="InputRadio-sm"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-primary radio-sm"
          data-testid="InputRadio"
          name="InputRadio-sm"
          type="radio"
        />
      </label>
    </div>
  </div>,
  <div
    className="form-control my-4 w-[200px]"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      xs
    </label>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 1
        </span>
        <input
          className="radio radio-primary radio-xs"
          data-testid="InputRadio"
          name="InputRadio-xs"
          type="radio"
        />
      </label>
    </div>
    <div
      className="form-control"
      data-testid="FormControl"
    >
      <label
        className="label"
        data-testid="Label"
      >
        <span
          className="label-text"
        >
          Option 2
        </span>
        <input
          className="radio radio-primary radio-xs"
          data-testid="InputRadio"
          name="InputRadio-xs"
          type="radio"
        />
      </label>
    </div>
  </div>,
]
`;
